ext/oj/dump.c

```diff
@@ -566,7 +566,7 @@ void oj_dump_xml_time(VALUE obj, Out out) {
             format[32] = '0' + out->opts->sec_prec;
             len -= 9 - out->opts->sec_prec;
         }
-        sprintf(buf, format, ti.year, ti.mon, ti.day, ti.hour, ti.min, ti.sec, (long)nsec);
+        len = sprintf(buf, format, ti.year, ti.mon, ti.day, ti.hour, ti.min, ti.sec, (long)nsec);
```

Comment:
I'm not sure which is best for determining the length. I suspect you change where Len is assigned to the return of sprintf is best which means the line above `len -= 9 - out->opts->sec_prec;` can be removed. No need to set Len to 30 either if using the sprintf return value.

Reply:
Thanks. I removed redundant codes.